xdsl/frontend/builder.py

```diff
@@ -0,0 +1,51 @@
+import ast
+from dataclasses import dataclass
+from typing import Any
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.frontend.pyast.code_generation import CodeGeneration
+from xdsl.frontend.pyast.utils.python_code_check import PythonCodeCheck
+from xdsl.frontend.pyast.utils.type_conversion import (
+    FunctionRegistry,
+    TypeConverter,
+    TypeRegistry,
+)
+from xdsl.transforms.desymref import Desymrefier
+
+
+@dataclass
+class PyASTBuilder:
+    """Build an operation ."""
+
+    type_registry: TypeRegistry
+    function_registry: FunctionRegistry
+
+    file: str
+    globals: dict[str, Any]
+    ast: ast.FunctionDef
+
+    desymref: bool
```

Comment:
Can you please add a one-line doc string for each of these (maybe a bit longer for desymref), and for the functions below?